### PR TITLE
Handle the case when no data is passed to charts

### PR DESCRIPTION
**Pull request by ioanungurean**
Added also a new webpack config so that the production size will be 4x smaller

### CODE REVIEW

File: src/graph/graph.js [hunk 1]
```diff
@@ -174,6 +176,35 @@ class Graph {
     });
   }
 
+  /**
+   * Checks if the parsed input data is not empty
+   * @returns {void}
+   */
+  checkForData() {
```
Comment (ntarocco) [inline src/graph/graph.js]: I would suggest to rename this to something like:
`hasData` or similar, and then when you call it:

```
if (!super.hasInputData()) { 
    super.renderNoData(); 
    return; 
}
```
In general, it is easier to have "positive" names without "not" and have the condition checking that "!hasData()"

**Review by ntarocco — APPROVED** [review]

File: src/graph/graph.js (hunk 1 above)
Comment (ioanungurean) [reply src/graph/graph.js]: That's a very good point. I updated the commit. :)

**Comment by ioanungurean** [comment]
My bad...

File: src/graph/graph.js [hunk 2]
```diff
@@ -152,15 +154,15 @@ class Graph {
...
-    Object.keys(this.input).forEach((k) => {
-      const obj = {};
-      obj.label = k;
-      obj.data = this.input[k].buckets;
-      parsed.push(obj);
-    });
-
-    this.input = parsed;
+    this.input = Object.values(this.input).reduce((acc, current) => {
+      if (Array.isArray(current.buckets) || current.buckets.length) {
```
Comment (ntarocco) [inline src/graph/graph.js]: I think it should `&&` and not `||`.

Input example:
```
{
    f: {buckets: [1,2,3]},
    s: {buckets: []},
    t: {buckets: [4,5,6]},
}
```

**Review by ntarocco — CHANGES_REQUESTED** [review]

**Comment by ioanungurean** [comment]
@ntarocco true, I totally missed that one

**Review by ntarocco — APPROVED** [review]

**Comment by ioanungurean** [comment]
Locally the tests are passing. Not sure why Travis is failing. 

I tried locally to change node version to match the Travis one and also packages versions to match the Travis one and the tests are still passing.

image

**Pull request merged**